### PR TITLE
chore(mantine): bumped to latest major and fixed / completed the migr…

### PR DESCRIPTION
**Pull request by FelixBlaisThon**
…ation

https://coveord.atlassian.net/browse/UITOOL-1209

### Proposed Changes

Bumped mantine to latest major and applied fixed / modification with respect to the new release

### Acceptance Criteria

-   [ ] The proposed changes are covered by unit tests
-   [x] The potential breaking changes are clearly identified
-   [ ] [README.md](https://github.com/coveo/plasma/blob/master/README.md) is adjusted to reflect the proposed changes (if relevant)

### CODE REVIEW

**Review by gdostie — COMMENTED** [review]
Good work! here is a few things I noticed in the demo

- Table `th` looks weird image
- The calendar style is broken image
- Modal wizard is broken 
image

File: packages/mantine/src/theme/Theme.tsx [hunk 1]
```diff
@@ -134,6 +151,15 @@ export const plasmaTheme: MantineThemeOverride = {
...
         },
+        DateInput: {
+            styles: {
+                root: {
+                    dropdown: {
+                        display: 'none',
+                    },
+                },
+            },
+        },
```
Comment (gdostie) [inline packages/mantine/src/theme/Theme.tsx:162]: I saw that we added that already in our table date pickers, do we need to add it in the theme as well?

File: packages/mantine/src/theme/Plasmantine.tsx [hunk 2]
```diff
@@ -1,10 +1,12 @@
 import {MantineProvider} from '@mantine/core';
+import {Notifications} from '@mantine/notifications';
 import {FunctionComponent, PropsWithChildren} from 'react';
 
 import {plasmaTheme} from './Theme';
 
 export const Plasmantine: FunctionComponent<PropsWithChildren> = ({children}) => (
     <MantineProvider withGlobalStyles withNormalizeCSS theme={plasmaTheme}>
+        <Notifications position="top-center" />
```
Comment (gdostie) [inline packages/mantine/src/theme/Plasmantine.tsx:9]: Not sure we really want to sneak in the notifications in there. It is not obvious to guess that the theme provider would do that.

Comment (FelixBlaisThon) [reply packages/mantine/src/theme/Plasmantine.tsx:9]: from the doc ```Add Notifications component anywhere in your application, usually it is placed inside [MantineProvider](https://mantine.dev/theming/mantine-provider/) next to your App component:```.  I will try to put it from the _app.tsx file and see how it goes

File: packages/mantine/src/theme/Theme.tsx (hunk 1 above)
Comment (FelixBlaisThon) [reply packages/mantine/src/theme/Theme.tsx:162]: I've added it for testing but forgot to remove it afterward. Consider it removed

File: packages/mantine/src/components/table/TablePagination.tsx [hunk 3]
```diff
@@ -24,19 +24,26 @@ export const TablePagination: FunctionComponent<TablePaginationProps> = ({totalP
...
             boundaries={0}
             size="md"
-            getItemAriaLabel={(label) => {
-                switch (label) {
-                    case 'prev':
-                        return 'previous page';
+            getItemProps={(page) => ({
+                component: 'button',
+                'aria-label': `${page}`,
+            })}
```
[the remark this answers is not in the document]
Comment (gdostie) [reply packages/mantine/src/components/table/TablePagination.tsx:35]: it seems to have an impact on the buttons radius

| current branch | master branch |
| --- | --- |
| image | image |

File: packages/mantine/src/theme/Plasmantine.tsx (hunk 2 above)
Comment (gdostie) [reply packages/mantine/src/theme/Plasmantine.tsx:9]: Oh I didn't know the documentation was recommending this. If its mentioned in the docs, then it becomes more obvious that the theme provider does that. You can ignore my comment I guess.

File: packages/mantine/src/components/table/TablePagination.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (FelixBlaisThon) [reply packages/mantine/src/components/table/TablePagination.tsx]: I found out it was useless, so I removed it

File: packages/style/gulpfile.js [hunk 4]
```diff
@@ -7,7 +7,6 @@
 
 const path = require('path');
 const del = require('del');
-const pngSprite = require('coveo-png-sprite');
```
Comment (GermainBergeron) [inline packages/style/gulpfile.js:10]: We can probably remove that dependency?

**Review by GermainBergeron — APPROVED** [review]

**Pull request merged**